blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/ScheduleContext.java

```diff
@@ -252,17 +264,23 @@ void scheduleEligibleSlots(long maxAgeMillis, long delayedMetricsMaxAgeMillis) {
         }
     }
 
-    private boolean areChildKeysOrSelfKeyScheduledOrRunning(SlotKey slotKey) {
+    boolean areChildKeysOrSelfKeyScheduledOrRunning(SlotKey slotKey) {
         // if any ineligible (children and self) keys are running or scheduled to run, we shouldn't work on this.
         Collection<SlotKey> ineligibleKeys = slotKey.getChildrenKeys();
 
-        if (runningSlots.keySet().contains(slotKey) || scheduledSlots.contains(slotKey)) {
+        if (runningSlots.keySet().contains(slotKey)) {
+            return true;
+        }
+        if (scheduledSlots.contains(slotKey)) {
             return true;
         }
 
         // if any ineligible keys are running or scheduled to run, do not schedule this key.
         for (SlotKey childrenKey : ineligibleKeys) {
-            if (runningSlots.keySet().contains(childrenKey) || scheduledSlots.contains(childrenKey)) {
+            if (runningSlots.keySet().contains(childrenKey)) {
+                return true;
+            }
+            if (scheduledSlots.contains(childrenKey)) {
```

Comment:
@ChandraAddala I understand the sentiment, and I'm not normally one for changing the _meaning_ of the code, under normal circumstances. But this doesn't change the logic at all. For all I know, _it will compile to exactly the same bytecode_. And if not, it will compile to trivial equivalent bytecode. It's less of a change than what jacoco does to instrument the code. Personally, I find it a little easier to read because it doesn't run off the right side of the screen.